bin/commonmark

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env php
+<?php
+
+requireAutoloader();
+
+ini_set('display_errors', 'stderr');
+
+foreach ($argv as $i => $arg) {
+    if ($i === 0) {
+        continue;
+    }
+
+    if (substr($arg, 0, 1) === '-') {
+        switch ($arg) {
+            case '-h':
+            case '--help':
+                help();
+                break;
+            default:
+                fail('Unknown option: ' . $arg);
+        }
+    } else {
+        $src = $argv[1];
+    }
+}
+
+if (isset($src)) {
+    if (!file_exists($src)) {
+        fail('File not found: ' . $src);
+    }
+
+    $markdown = file_get_contents($src);
+} else {
+    $stdin = fopen('php://stdin', 'r');
+    stream_set_blocking($stdin, false);
+    $markdown = stream_get_contents($stdin);
+    fclose($stdin);
+
+    if (empty($markdown)) {
+        help();
```

Comment:
help output may better go to STDERR instead of STDOUT to have empty output on stdout when input is empty.